internal/commands/secrets/delete.go

```diff
@@ -0,0 +1,89 @@
+package secrets
+
+import (
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+
+	"github.com/spf13/pflag"
+)
+
+// CommandDelete for the secrets delete command
+type CommandDelete struct {
+	inputs      deleteInputs
+	realmClient realm.Client
+	outputs     secretOutputs
+}
+
+// Inputs function for the secrets delete command
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Flags function for the secrets delete command
+func (cmd *CommandDelete) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+	fs.StringSliceVarP(&cmd.inputs.secrets, flagSecret, flagSecretShort, []string{}, flagSecretUsage)
+}
+
+// Setup function for the secrets delete command
+func (cmd *CommandDelete) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler function for the secrets delete command
+func (cmd *CommandDelete) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+
+	secretList, secretListErr := cmd.realmClient.Secrets(app.GroupID, app.ID)
+	if secretListErr != nil {
+		return secretListErr
+	}
+
+	toDelete, resolveErr := cmd.inputs.resolveSecrets(ui, secretList)
```

Comment:
NOTE: once @kpatel71716 merges his `user revoke` work, we'll have an established pattern for commands structured like this going forward.  i think it'll be worth your time here to make the necessary refactors to line up with that pattern, so going forward `users` AND `secrets` can be considered "scope-complete"